contrib/OtlpGrpc/Exporter.php

```diff
@@ -0,0 +1,184 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OpenTelemetry\Contrib\OtlpGrpc;
+
+use grpc;
+use Opentelemetry\Proto;
+use Opentelemetry\Proto\Collector\Trace\V1;
+use Opentelemetry\Proto\Common\V1\InstrumentationLibrary;
+use Opentelemetry\Proto\Trace\V1\InstrumentationLibrarySpans;
+use OpenTelemetry\Sdk\Trace;
+use OpenTelemetry\Trace as API;
+
+use OpenTelemetry\Trace\Span;
+
+class Exporter implements Trace\Exporter
+{
+    /**
+     * @var string
+     */
+    private $endpointURL;
+
+    /**
+     * @var string
+     */
+    private $protocol;
+
+    /**
+     * @var string
+     */
+    private $insecure;
+
+    /**
+     * @var string
+     */
+    private $certificateFile;
+
+    /**
+     * @var array
+     */
+    private $headers;
+
+    /**
+     * @var string
+     */
+    private $compression;
+
+    /**
+     * @var int
+     */
+    private $timeout;
+    /**
+     * @var SpanConverter
+     */
+    private $spanConverter;
+    
+    /**
+    * @var bool
+    */
+    private $running = true;
+
+    /**
+     * @var ClientInterface
+     */
+
+    private $client;
+
+    /**
+     * OTLP GRPC Exporter Constructor
+     * @param string $serviceName
+     */
+    public function __construct(
+        string $endpointURL = 'localhost:4317',
+        bool $insecure = true,
+        string $certificateFile = null,
+        array $headers = [],
+        bool $compression = false,
+        int $timeout = 10,
+        ClientInterface $client = null
+    ) {
+
+        // Set default values based on presence of env variable
+        $this->endpointURL = getenv('OTEL_EXPORTER_OTLP_ENDPOINT') ?: $endpointURL;
+        $this->protocol = getenv('OTEL_EXPORTER_OTLP_PROTOCOL') ?: 'grpc'; // I guess this is redundant?
+        $this->insecure = getenv('OTEL_EXPORTER_OTLP_INSECURE') ?: $insecure;
+        $this->certificateFile = getenv('OTEL_EXPORTER_OTLP_CERTIFICATE') ?: $certificateFile;
+        $this->headers[] = getenv('OTEL_EXPORTER_OTLP_HEADERS') ?: $headers; // TODO
+        $this->compression = getenv('OTEL_EXPORTER_OTLP_COMPRESSION') ?: $compression; // TODO
+        $this->timeout =(int) getenv('OTEL_EXPORTER_OTLP_TIMEOUT') ?: $timeout; // TODO
+
+        $this->spanConverter = new SpanConverter('foo');
```

Comment:
Other exporters have passed the service name (`foo` in this case) in via the constructer but it can also be set from `OTEL_RESOURCE_ATTRIBUTES`

Reply:
So I've looked into this a little more, I see that most of the PHP Exporters take in the `serviceName`. So I went down the rabbit hole of `OTEL_RESOURCE_ATTRIBUTES` in other languages, and they seem to pass in `service.name` etc into a `Resource` on the `TracerProvider` rather than passing the service name via the Exporter. See these two examples:

https://github.com/lightstep/opentelemetry-examples/blob/main/go/opentelemetry/server/server.go#L116
https://github.com/honeycombio/examples/blob/main/python-otlp/app.py#L20

Which is swaying me more to removing the `$serviceName` from the Exporter entirely. What are your thoughts?

Reply:
If `OTEL_RESOURCE_ATTRIBUTES` exists it is prioritized; if it doesn't exist, then the `service.name` is set in the [tracer](https://github.com/open-telemetry/opentelemetry-php/blob/main/sdk/Trace/TracerProvider.php#L70) and should be pulled from there to set the service name for export.

You can get the `ResourceInfo` from the tracer then iterate through the keys and set the attributes on the span like python. 
https://github.com/open-telemetry/opentelemetry-python/blob/main/exporter/opentelemetry-exporter-otlp-proto-grpc/src/opentelemetry/exporter/otlp/proto/grpc/exporter.py#L127